subsys/zbus/zbus.c

```diff
@@ -380,10 +380,8 @@ int zbus_chan_pub(const struct zbus_channel *chan, const void *msg, k_timeout_t
 
 	_ZBUS_ASSERT(chan != NULL, "chan is required");
 	_ZBUS_ASSERT(msg != NULL, "msg is required");
-
-	if (k_is_in_isr()) {
-		timeout = K_NO_WAIT;
-	}
```

Comment:
Please do not remove it. We use semaphore internally, and a semaphore cannot receive a timeout different from `K_NO_WAIT` inside ISRs. So, the assert is very welcome, but removing the `if` would make the code less robust.